core-common/src/main/java/org/mapstruct/IterableMapping.java

```diff
@@ -52,4 +55,11 @@
      * @return the qualifiers
      */
     Class<? extends Annotation>[] qualifiedBy() default { };
+
+    /**
+     * Specifies the result type of the element mapping method to be used in case multiple mapping methods qualify.
+     *
+     * @return the resultType to select
+     */
+    Class<?> resultType() default void.class;
```

Comment:
`resultType` seems confusing to me, as it's really the type of the elements. Maybe `elementTargetType`, `keyTargetType`, `valueTargetType` (for map mappings) and `targetType` for bean mappings? I liked result type for the latter, but it doesn't seem to scale for iterables/maps.